### ISSUE TITLE
Intermittent failures with certain amd64 images when using > 1 CPU (Apple M1)

### ISSUE THREAD

**Issue by lyuanxi** [docker/for-mac#6016]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 10CAF6D5-EB01-419D-BAEE-0865A2CED4BE/20211022005141

### Expected behavior
Able to reliably use amd64 images (such as flyway) with multiple CPU (More than 1) configured in Docker Preferences - Resources. I've noticed this issue on at least flyway but there may be others.

### Actual behavior
With CPU set to 2 or more, images that would otherwise work normally start exhibiting intermittent failures such as randomly "hanging" until forced killed or ocassionally with a qemu error. My use case is when using flyway migration to execute sql statements on a postgres database (Also using Docker)

### Information

  - macOS Version: Big Sur 11.6
  - Intel chip or Apple chip: Apple Chip
  - Docker Desktop Version: 4.1.1

### Steps to reproduce the behavior

My project uses these 2 images (amd64)
- `docker pull postgres:13.2-alpine`
- `docker pull flyway/flyway:7.8.2-alpine`

We run a postgres database and use flyway for migrations. There are over 500 migration scripts currently. When Docker Preference - Resources is set to **2 CPU or more**, one of these occur:
1. It runs successfully without issues
2. It fails to attach flyway and "hangs" and needs to be force killed
3. Flyway starts running but intermittently "hangs"  at random points, no apparent pattern. Needs to be force killed.
4. It crashes with the following error: 
```sh
flyway-dev | qemu: uncaught target signal 5 (Trace/breakpoint trap) - core dumped
flyway-dev | ./flyway: line 83: 50 Trace/breakpoint trap "$JAVA_CMD" $JAVA_ARGS $EXTRA_ARGS -cp "$CP" org.flywaydb.commandline.Main "$@"
```

When **CPU is configured to be 1**, the above issues no longer occur. This is not ideal because performance is impacted. This highlights a multi-CPU issue of sorts when running amd64 images on Apple Chips which is inconvenient when arm64 images are not available.

Other things of note:
Flyway currently does not officially support a arm64 images however when I built my own arm64 image off the official repository (https://github.com/flyway/flyway-docker) with `docker buildx build --platform=linux/arm64 .` and use that instead. The above multi-CPU issues appear to be resolved.

**Comment by xoxwgys56** [comment]
@longwa Could you share how to set single core for QUEMU?  

I thought you said about `docker desktop`. so I set 1 Cpu on resources tab. but it did not work.

**Comment by longwa** [comment]
Setting to 1 core in docker desktop should work around the issue assuming they don't have any other QEMU settings that are causing problems. Some software (such as Oracle db for instance) still doesn't seem to work but it does fix the random hangs I was seeing with a few things (running Java, for instance).

I think the docker desktop team will need to modify their code to change the settings for QEMU. I'm assuming they are using the Force Multicore option since I can see Docker Desktop using more than 100% CPU on my M1.

**Comment by Jdban** [comment]
Does anyone have a good solution to this? I'm still having the issue when docker is set to 1 CPU.

**Comment by jedisct1** [comment]
@Jdban I eventually switched to using Lima (via [colima](https://github.com/abiosoft/colima)).

The switch was painless and it works perfectly fine on my M1.

**Comment by Jdban** [comment]
> @Jdban I eventually switched to using Lima (via [colima](https://github.com/abiosoft/colima)).
> 
> The switch was painless and it works perfectly fine on my M1.

Thanks for the suggestion @jedisct1. I tried out colima and it has been super useful. I had to make some changes to the lima override.yaml for my setup but it worked out.

I'm still getting VERY intermittent hangs, so hopefully I can figure out what's going on in the future. Today I had 214 docker containers start up and complete before the 215th hung and had to be killed. Definitely odd, but I can work with that I guess. 

Thanks!
